// Copyright (c) 2020 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api;

option java_outer_classname = "PowerServiceProto";

import "bosdyn/api/power.proto";

service PowerService {
    // Starts a power command on the robot. A robot can only accept one power command at once.
    // Power commands, are not interruptible. Once a command is issued, it must complete before
    // another command can be issued.
    rpc PowerCommand(PowerCommandRequest) returns (PowerCommandResponse) {}

    // Check the status of a power command.
    rpc PowerCommandFeedback(PowerCommandFeedbackRequest) returns (PowerCommandFeedbackResponse) {}
}
